SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -608,11 +608,18 @@ int64 USpatialActorChannel::ReplicateActor()
 
 			bCreatedEntity = true;
 
-			// Since we've tried to create this Actor in Spatial, we no longer have authority over the actor since it hasn't been delegated to us.
-			if (!USpatialStatics::IsSpatialOffloadingEnabled())
+			// If we're spawning an Actor that we know will be load-balanced to another worker,
+			// then preemptively set the role to SimulatedProxy.
+			if (SpatialGDKSettings->bEnableUnrealLoadBalancer)
 			{
-				Actor->Role = ROLE_SimulatedProxy;
-				Actor->RemoteRole = ROLE_Authority;
+				const UAbstractLBStrategy* LBStrategy = NetDriver->LoadBalanceStrategy;
+				if (!LBStrategy->ShouldHaveAuthority(*Actor))
+				{
+					Actor->Role = ROLE_SimulatedProxy;
+					Actor->RemoteRole = ROLE_Authority;
+
+					UE_LOG(LogSpatialActorChannel, Log, TEXT("Spawning Actor that will immediately become authoritative on a different worker. Actor: %s. Target virtual worker: %d"), *Actor->GetName(), LBStrategy->WhoShouldHaveAuthority(*Actor));
```

Comment:
Are we hoping that this doesn't happen in the long term / are there incoming changes where this flow isn't correct? I am guessing this is for the sake of metrics or smth?

Reply:
It's fine to do, hence being Log and not Warning, but it's likely there are a fair few issues that will arise from unintended immediate migrations and I think being able to quickly identify relevant whether an Actor is immediately handed over from logs will be useful debugging.